fix(sample): Add sample_event tag to samples

PixelDust22 opened the pull request:
This allows us to tell if an event is a sample event more reliably.

PixelDust22 committed:
fix(sample): Set sample_event tag on sample events

PixelDust22 committed:
fix(sample): Add sample_event tag to samples

scefali commented on src/sentry/utils/samples.py:
```diff
@@ -301,10 +301,14 @@ def create_sample_event(
 
 
-def create_sample_event_basic(data, project_id, raw=True, skip_send_first_transaction=False):
+def create_sample_event_basic(
```
@Neo-Zhixing instead of adding a new argument, can't we just modify `data` as we pass it in?

Neo-Zhixing replied:
Sure, but then you'll have to do this from two places

scefali replied:
@Neo-Zhixing that's fine, we can just make a function to do it so we have less repeating code

Neo-Zhixing replied:
Both from `create_sample_event`, and the create transaction endpoint, because the create transaction endpoint calls `create_sample_event_basic` while the create error endpoint calls `create_sample_event`.

scefali replied:
@Neo-Zhixing For transactions, we have a function called `fix_event_data` which is the best place to make modifications for sample event info. We don't have something for sample errors but maybe with refactoring, we can make it work in a similar way.

Neo-Zhixing replied:
To me the `sample_event` tag is clearly coupled with the fact that it was created by `create_sample_event_basic`. So it makes sense to do this in `create_sample_event_basic` as an argument. `fix_event_data` does one thing which is fixing event data, but with the `sample_event` flag you're not fixing things, you're adding onto the data.

scttcper committed:
feat(alertStatus): Add list of issues to the alert status page (#31930)

scefali commented on src/sentry/utils/samples.py:
```diff
@@ -301,10 +301,14 @@ def create_sample_event(
 
     data.update(kwargs)
-    return create_sample_event_basic(data, project.id, raw=raw)
+    return create_sample_event_basic(data, project.id, raw=raw, tagged=True)
```
@Neo-Zhixing I realize we use the same code path for load mocks as well which probably shouldn't be tagged this way

Neo-Zhixing replied:
I'll add a new argument to `create_sample_event` as well and pass this down to `create_sample_event_basic`

evanpurkhiser committed:
feat(pageFilters): Store desynced filters in store (#31985)

PixelDust22 committed:
add tagged argument to create_sample_event

mikellykels committed:
fix(alertWizard): Omit filter tags not relevant to event dataset (#32014)

Omit filter tags that are not relevant to the event dataset.

FIXES WOR-1589

JonasBa committed:
ref(tsc): improve locale types (#31132)

* ref(tsc): improve locale types

* fix(locale): narrow down the type to string

evanpurkhiser committed:
chore(js): Upgrade framer-motion -> 6x (#32013)

JonasBa committed:
feat(flamegraph): fix recursion coloring (#31948)

* feat(flamegraphs): add thread selector

feat(flamegraphs): add thread selector

* fix(flamegraphtoolbar): export toolbar directly

* feat(flamegraph): create preferences hook

* fix(flamegraph): add flamegraph view

* ref(flamegraph): change constructor signature

* feat(flamegraph): lift flamegraph preferences to provider and add color coding

* feat(flamegraph): improve color encoding calls

* fix(flamegraph): add preferences

* fix(flamegraph): fix tests and types

* fix(flamegraph): fix js profile tests

* fix(flamegraph): fix sampled profile tests

* fix(flamegraph): prevent minimap from shrinking

shruthilayaj committed:
feat(dashboard): Add aggregates and columns fields (#31953)

Add aggregates and columns fileds to DashboardWidgetQuery
django model to store these values in two separate fields
instead of a single fields field.

JonasBa committed:
fix(globalsdkupdates): refactor to fn component and rtl tests (#31667)

* fix(globalsdkupdates): refactor to fn component and rtl tests

* ref(globalsdkupdatealet): avoid array copy

* test(globalsdkupdatealet): silence act errors

* test(globalsdkupdatealet): use mockapiclient

* fix(globalsdkupdatealert): simpler and better tests

* test(globalSdkUpdateAlert): drop OrganizationContext

* test(globalSdkUpdateAlert): drop OrganizationContext in projectDetails

Co-authored-by: Jonas Badalic <jonasbadalic@Jonass-MacBook-Pro.local>

JonasBa committed:
ref(button): improve tsc compile time (#31699)

* ref(button): use interface for props

* ref(button): adopt the new conditional aria type

* ref(button): export both props

* ref(button): fix styled prop

* ref(button): modify imports

* fix(button): revert old changes

* fix(button): relax children type

* fix(button): aria label casing

* Update static/app/components/confirm.tsx

Co-authored-by: Priscila Oliveira <priscila.oliveira@sentry.io>

* Update static/app/components/confirm.tsx

Co-authored-by: Priscila Oliveira <priscila.oliveira@sentry.io>

* style(lint): Auto commit lint changes

* style(lint): Auto commit lint changes

* ref(components): use withoutaria props

* fix(button): omit prop at the right place

* fix(addintegrationbutton): remove child

Co-authored-by: Priscila Oliveira <priscila.oliveira@sentry.io>
Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

narsaynorath committed:
feat(new-widget-builder-experience): Additional error handling in builder (#32012)

dependabot[bot] committed:
chore(deps): update sentry-sdk requirement from <1.5.0,>=1.4.3 to >=1.4.3,<1.6.0 (#31432)

wmak committed:
feat(MEP): Add epm & failure rate to the query builder (#32003)

- This adds both the EPM(along with EPS) and failure rate to the metrics
  query builder
- This moves the non failure statuses to the constants file so that
  discover & metrics can share them

Zylphrex committed:
ref(profiling): Cleanup flamegraph renderering logic (#32018)

Previously, most of the logic was handled within a single useEffect. This change
breaks the useEffect into several parts.

1. Use useMemo to create all the renderers
2. Use separate useEffect to register the draw call for each canvas
3. Use an useEffect to register all the listeners on the scheduer
4. Use an useEffect to watch the canvas for any resize changes
5. Use an useEffect to register the scheduler and trigger the initial draw

This achieves a few things

- logical separation between where each canvas is being drawn
- remove the hack where the searchResults was explicitly changing drawRectangles
- remove extra draw calls to the overlay around the selectedFrameRenderer

JoshFerge committed:
chore(dx): make api-owners less noisy (#32028)

JoshFerge committed:
feat(apidocs): validate description exists (#31988)

* feat(apidocs): validate description exists

* fix typing

* improve check_tag function

* fix typing

Zylphrex committed:
feat(profiling): Fetch profiles list from api (#31971)

This fetches the profiles list from the api and populate scatter plot and table
on the profiling page.

edwardgou-sentry committed:
feat(dashboards): Issue Widget annotations field (#31670)

Adds support to display Issue Widget annotations as a column for use cases such as Jira links.

wedamija committed:
feat(issue_alert_status): Add backend abstraction for accessing issue alert rule firing history (#32015)

Adding a backend abstraction here because we likely want to move this into ClickHouse at some point.
Initially just including `record` in the interface, but we'll also need various methods for fetching
data that I'll define once get into the implementation.

vuluongj20 committed:
fix(dashboards): Use title styles for general dashboard title (#32039)

markstory committed:
chore: Remove unused feature flags (#32010)

These feature flags had no checks in either sentry or getsentry and are
just adding overhead to Organization Details response times.

I'll remove these flags from Flagr as well.

JoshFerge committed:
feat(apidocs): type team serializer response (#32029)

taylangocmen committed:
fix(ui): Fix thresholds section title in alert details (#32030)

wedamija committed:
typing: Add type hints to some base model files (#32024)

Added some basic typing here. It's hard to type this section well without using django type hinting
stubs, hopefully we can make those work in the future.

shruthilayaj committed:
fix(discover): Reintroduce sort on equation cols (#32027)

Updating column.name here https://github.com/getsentry/sentry/pull/31475
regressed the sort order. Using column.key to determine the fields
for sort instead.

wmak committed:
ref(mep): Move setup for storing metrics to a testutil case (#32038)

- This is so the code is easier to re-use for endpoint tests

mgaeta committed:
ref(jira): Split up long function (#31813)

mgaeta committed:
fix(plugins): Silence error (#32042)

robinrendle committed:
fix(ui): removed feedback alert on notifications page (#32043)

priscilawebdev committed:
feat(new-widget-builder-experience): Port create widget from discover experience - (#31957)

narsaynorath committed:
fix(new-widget-builder-experience): Add to New dashboard (#32055)

shruthilayaj committed:
feat(dam): Metrics tag store (#32035)

Store metrics tags in reflux store instead of
making an API request every time to retrieve
the data.

dashed committed:
fix(performance): tableCursor should not persist when switching tags (#32053)

k-fish committed:
ref(sdk): Add temp long task check inside VCD (#32072)

* ref(sdk): Add temp long task check inside VCD

This adds a long-task observer inside the visuallyCompleteData component that will show long tasks while the component is mounted. Adding this will hopefully shed some light on LCP measurements that are inexplicably longer than VCD measurements when no other spans exists.

narsaynorath committed:
feat(new-widget-builder-experience): Add widget library sidebar (#32021)

Zylphrex committed:
feat(profiling): Make each point on scatter plot clickable (#32047)

The scatter plot should also serve as a method of navigation. This change allows
clicking on a data point on the scatter plot in order to navigate to the
corresponding flamegraph.

AniketDas-Tekky committed:
ref(metrics): Track if a request is from the frontend (#32058)

* ref(metrics): Track if a request is from the frontend

This change checks if an incoming request has `COOKIES`. If so, we'll mark it as such in metrics logging and the access logs

* Fixed failing tests

wedamija committed:
types: Add type hints to some database fields (#32025)

There's a fair bit to do to get all the fields types, just did some of the work for now.

wedamija committed:
feat(issue_alert_status): Add model to keep track of issue alert rule firing history (#32040)

This model will be used to store alert rule firing history in Postgres. We'll use this together with
a Postgres specific backend to power the issue alert rule status page.

dashed committed:
feat(performance): Add dropdown menu to web vitals content page (#32057)

Co-authored-by: Ash Anand <0Calories@users.noreply.github.com>

mgaeta committed:
fix(jira): Set Badge (#32054)

narsaynorath committed:
meta(editor): Set frontend language tab sizes to 2 (#32071)

maheskett committed:
feat(db): Support dict of params for raw SQL queries (#31873)

k-fish committed:
feat(perf): Add anomalies backend and summary graphs (#32050)

* feat(perf): Add anomalies backend and summary graphs

This hooks up the actual anomalies backend and adds anomalies to the transaction summary tpm graph.

This is an pre-internal PR for the anomalies ui, still behind a feature flag, to get this onto prod for others to test before internal.
Still needs some extra work on tests etc. but this should be enough to play with for now.

edwardgou-sentry committed:
fix(dashboards): Widget Viewer World Map table should always display geo.country_code (#32037)

Updates the World Map Table for the Widget Viewer to always include geo.country_code as a column.

vuluongj20 committed:
feat(dropdown-menu-v2): Add min width to menu (#32061)

* feat(dropdown-menu-v2): Add min width to menu

* ref(dropdown-menu-v2): Use useResizeObserver to update triggerWidth

Commit:
feat(codeowners): Show CTA to all orgs with active Stacktrace Linking (#31961)

wmak committed:
fix(mep): Properly calculate granularity & interval for timeseries (#32052)

- This fixes how the `time` column and granularity are determined for
  timeseries queries
  - In these cases, we don't want to just select `timestamp` as-is, but
    instead round them to the "interval" that the frontend is selecting.
    And then use the same logic as the MetricQueryBuilder to determine
    the correct granularity to use
  - This is because if we set granularity == interval, there will be
    events that we'll miss since granularities because of the way
    buckets are aligned

beezz committed:
Use replica to get the ids for deletion (#32068)

markstory committed:
fix(lint) Fix errors & warnings for upcoming changes in eslint config (#32079)

Refs getsentry/eslint-config-sentry#131

dashed committed:
fix(perf): Fix error message in span view (#32091)

evanpurkhiser committed:
chore(js): Upgrade fuze -> 6.0 (#32060)

Commit:
fix(codeowners): Fix CTA flashing on initial render (#32096)

dashed committed:
chore(perf): Move "All Events" tab to be the second tab (#32083)

PixelDust22 committed:
fix(report): Weekly email copy / style / query update (#32095)

* css fixups

* copy changes

* update snql query

* Revert "update snql query"

This reverts commit a57c0425c4ee5f310e432c81ed7349d2444ed871.

* Revert "copy changes"

This reverts commit ace1d5f9dbb8b0bd0216836a44c4695c706b333b.

* updates

jjbayer committed:
feat(releasehealth): Track delta distribution as metric and tag (#32036)

Record each numeric delta between sessions and metrics results as a
histogram metric and a sentry tag (rounded to percentage).

JonasBa committed:
ref(tests): drop dead code (#32067)

* ref(tests): drop dead code

* ref(tests): more dead code

JonasBa committed:
feat(flamegraph): add js self profiling example (#32062)

* feat(flamegraph): add js self profiling example

* fix(jsselprofiling): do not inject native code for market frames

* ref(filename): camelcase everything (#32063)

* ref(filename): camelcase everything

* ref(filename): camelcase everything

* ref(filename): camelcase everything

* ref(filename): drop diff

* add back theme files

* feat(profile): extend importProfile to support raw profile trace

jjbayer committed:
fix(releasehealth-duplex): Log strings and prevent division by zero (#32113)

Clean up after #32036.

sl0thentr0py committed:
ref(sdk): Remove unused on_full_queue handler in sdk (#32115)

Was removed in the SDK in https://github.com/getsentry/sentry-python/pull/1100

narsaynorath committed:
fix(new-widget-builder-experience): Required indicator and error unset (#32077)

Zylphrex committed:
chore(profiling): Add profiling table to story book (#32076)

narsaynorath committed:
fix(new-widget-builder-experience): Editing widgets (#32080)

Switches to using the index in the URL because new widgets do not have
an id. Also passes down the correct update handler depending on edit state.

scttcper committed:
test(alerts): Speed up create alert rule test (#32103)

shruthilayaj committed:
feat(dam): Metrics meta reflux store (#32069)

Adds metrics meta to reflux store.

Zylphrex committed:
feat(profiling): Customize profiling scatter chart tooltip (#32084)

The tooltip currently only shows the timestamp and the duration of each profile.
This change adds more information to the tooltip from the profile.

narsaynorath committed:
feat(new-widget-builder-experience): Add border between form and sidebar - [DD-536] (#32073)

JoshFerge committed:
feat(apidocs): resolve TYPE_CHECKING imports while building api docs (#31958)

Commit:
feat(sentryapp): Sentry apps request with dependentData (#31882)

Objective:
This PR adds the dependentData query param for fields with a depends_on property. We also make the requests with the dependentData query param when loading a saved rule.

vuluongj20 committed:
fix(dropdown-menu-v2): Fix ResizeObserver loop limit error (#32100)

* fix(dropdown-menu-v2): Fix ResizeObserver loop limit error

* style: Use promise syntax for infinite loop fix

mikellykels committed:
fix(alertWizard): Omit transaction tags for event dataset filter (#32026)

Omit transaction tags for event dataset filter.

FIXES WOR-1655

mikellykels committed:
feat(alertWizard): Remove legend and make labels consistent with Alerts graph (#32022)

Remove the legend because we already have the aggregate in the label. Also update the chart labels to be consistent with the Alerts graph labels.

FIXES WOR-1591

scttcper committed:
feat(workflow): Remove issue-percent-filters flag (#32011)

markstory committed:
chore(ui) Add role to improve selectors in tests (#32119)

Add a role so we can select the ring as an image without having to use
`container.querySelector`

Refs getsentry/getsentry#7087

scttcper committed:
feat(alertStatus): Add pagination to issues table (#32019)

RyanSkonnord committed:
ref(access): Use OrganizationGlobalAccess in more cases (#32020)

Use the OrganizationGlobalAccess subclass (replacing concrete Access
objects) to represent where superusers and Sentry apps have access to
all an organization's teams and projects. Consistently expose those sets
of teams and projects as attributes (previously done for Sentry apps,
but not the superuser or global auth case).

Filter the `teams` attribute by visibility for all Access objects, for
consistency with OrganizationMember.get_teams (used by from_member).

RyanSkonnord committed:
fix(auth): Remove Okta migration workaround (#31800)

Delete "organizations:sso-migration" feature flag and associated logic.

mgaeta committed:
fix(jira): Webhook Response (#32109)

mgaeta committed:
fix(api-docs): Correct scope (#32093)

edwardgou-sentry committed:
feat(dashboards): hides the widget viewer button when in edit mode (#32122)

Hides the Widget Viewer button when in edit mode

scttcper committed:
test(alertStatus): Add missing context to test for router (#32135)

Commit:
feat(sentryapp): Fix edit view for SentryAppExternalForm (#31881)

Objective:
This fixes the logic for disabled state for fields with depends_on property during an edit view.

evanpurkhiser committed:
chore(js): Re-export Fuse type in utils/fuzzySearch (#32094)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

JonasBa committed:
feat(profiling): fetch flamegraph data on route (#32064)

* feat(flamegraph): add js self profiling example

* fix(jsselprofiling): do not inject native code for market frames

* ref(filename): camelcase everything

* ref(filename): camelcase everything

* ref(filename): camelcase everything

* ref(filename): drop diff

* add back theme files

* feat(flamegraph): add data loading

* fix(flamegraph): set proper activeProfileIndex index

* fix(flamegraph): remove unnecessary selection

* fix(flamegraph): relative container + center flamegraph

ahmedetefy committed:
ref: Remove unused variables (#32152)

shruthilayaj committed:
chore(perf): Remove references to legacy APDEX, USER_MISERY descriptions (#32125)

Remove references to legacy apdex.

shruthilayaj committed:
chore(perf): Remove frontend references to organization level apdex (#32118)

Orgs haven't been able to set org level apdex
thresholds for the last 6 months or so. Removing
remaining frontend references to
organization.apdexThreshold.

Zylphrex committed:
fix(trace-view): The query limit on trace view should be 100 (#32138)

The query limit was originally 100, but was dropped to 50 unintentionally during
the snql migration.

edwardgou-sentry committed:
feat(dashboards): Shareable links for widgetviewer (#31975)

Opening the Widget Viewer on a widget updates the browser url to include the widgetId being opened.
The updated url can be shared to directly open the Widget Viewer when visiting the url.

Zylphrex committed:
chore(profiling): Lift PerformanceDuration component (#32117)

Previously, the PerformanceDuration component was only being used in the
performance product. With the introduction of profiling, PerformanceDuration
will be used in more places as the default component for displaying duration
values. This change lifts the component into the top level components to avoid
profiling having to import from performance views.

Zylphrex committed:
chore(profiling): Use Pagefilter[datetime] on profiling chart (#32129)

wmak committed:
feat(mep): Connect the stats endpoint to metrics (#32082)

- This fixes a bug with the get_features locally not recognizing the
  metrics feature because of how batch_has doesn't check the settings
- This adds the metric referrers (ie. performance widget referrers) as
  accepted ones
- This adds a metrics_enhanced_performance file that will match closly
  to how discover works so that we can run queries either as plain
  discover or mep queries
- This hasn't added mep to the response meta yet, will do that in a
  follow up PR

edwardgou-sentry committed:
feat(dashboards): Widget Viewer style improvements and use GridEditable component (#32051)

Changes to some of the style in the Widget Viewer.
Also updates the Widget Viewer to use the GridEditable component to enable resizing headers.

This change also allows us to enable sort by clicking column headers, which will be completed in another PR.
Pagination is also todo

scttcper committed:
deps(ui): Upgrade jest, jest-junit (#32132)

priscilawebdev committed:
ref(new-widget-builder-experience): Port unit/integration tests - (#32070)

edwardgou-sentry committed:
fix(dashboard): Widget Viewer truncate widget title and fixed button size (#32123)

truncates widget viewer title and forces widget viewer button not to resize on long titles

markstory committed:
fix(ui) Fix overflow numbers in key stats (#32163)

As well as other minibarcharts with y-axis labels. This aligns the
y-axis formatting with organization stats and smaller charts in
performance.

wedamija committed:
feat(issue_alert_status): Start writing issue alert history (#32045)

This adds a postgres backend for issue alert history and calls record after an issue alert fires.

k-fish committed:
ref(perf): Fix VCD timestamps (#32168)

Start timestamp for the transaction changes, it's correct within the register before finish callback, so moving the calculations for VCD milliseconds inside the callback seems to work correctly.

MeredithAnya committed:
ref(metrics-indexer): Delete old batching metrics consumer (#31928)

k-fish committed:
feat(perf): Fix anomaly chart yAxis (#32170)

Was using p50 to coerce, switch to tpm so it's count. Can change this in the future if we do dynamic fields.

JoshFerge committed:
fix(apidocs): use datetime type instead of str (#32130)

vuluongj20 committed:
fix(dropdown-menu-v2): Fix submenus detached from their parents (#32121)

doralchan committed:
make btn size consistent on widgets (#32131)

mikellykels committed:
fix(alertStatus): Update font size and weight for alert conditions (#32156)

Update the font size and weight for alert conditions side bar

FIXES WOR-1662

volokluev committed:
feat(ratelimits): Limit by group as opposed to by endpoint by default (#32133)

* rate limit by group by default

* add test

* more descriptive typing

Co-authored-by: Vlad Kluev <vlad.kluev@sentry.io>

dashed committed:
fix(perf): Normalizing search conditions on performance product (#32145)

ceorourke committed:
ref(github plugin): Demote logger.error to logger.info (#32172)

priscilawebdev committed:
ref(new-widget-builder-experience): Port unit/integration tests - Part 2 - (#32155)

Zylphrex committed:
perf(issues): Instrument some possible performance bottle necks (#32175)

We're seeing some missing instrumentation around these areas in prod with a
noticeable performance impact on the overall request. This adds some
instrumentation to help us narrow down the culprit.

shruthilayaj committed:
chore(perf): Remove backend references to Organization Apdex Threshold (#32120)

Removes backend references to organization apdex.

ceorourke committed:
ref(bitbucket): Demote logger.error to logger.info (#32179)

Commit:
feat(page-filters): Add function to revert to pinned filters (#32140)

Used for when this revert button is clicked: #32141

wedamija committed:
perf(db): Count hits on a replica instead of the primary postgres (#32178)

This can be an expensive query, and it doesn't need to be 100% accurate, so we can safely use a
replica here. Replag is typically low anyway, so this shouldn't have any noticeable customer impact.

edwardgou-sentry committed:
fix(dashboards): Widget Viewer edit button regression fix (#32182)

Edit button in the Widget Viewer stopped working since we added shareable links.
This fixes by adding an onEdit handler to detail.tsx

AbhiPrasad committed:
build(js): Upgrade JS SDK to 6.18.1-beta.0 (#32181)

https://github.com/getsentry/sentry-javascript/pull/4531

This will also fix the console array problem.

wedamija committed:
feat(issue_alert_status): Add method to fetch issue alert group break down in a time range. (#32099)

This adds `fetch_rule_groups_paginated` to the `RuleHistoryBackend` classes. This returns a
`CursorResult` which contains a list of `RuleGroupHistory` objects as the result. These objects
encapsulate a `Group` and any extra context about them, which at the moment is just the number of
times the group triggered the rule.

dashed committed:
fix(discover): Fix docs link on Edit Column modal (#32183)

evanpurkhiser committed:
ref(alerts): Extract FilterBar component (#32185)

Co-authored-by: getsantry[bot] <66042841+getsantry[bot]@users.noreply.github.com>

beezz committed:
perf(db): Use read replica for snuba group candidates fetching (#32184)

wedamija committed:
fix(py): Bad import #32190

CI missed this because the import was in tests, and ci has responses installed. But this fails in
production since it isn't.

JoshFerge committed:
feat(orgstats): document v2 API (#32128)

Co-authored-by: Alberto Leal <mail4alberto@gmail.com>

PixelDust22 committed:
Merge branch 'fix/sample-event-tag' into master

Neo-Zhixing commented:
Closed due to issue when merging upstream
